gdal/frmts/netcdf/netcdfmultidim.cpp

```diff
@@ -1962,12 +1965,14 @@ bool netCDFVariable::IReadWriteGeneric(const size_t* arrayStartIdx,
                 if( (--nIters) == 0 )
                     break;
                 stack_ptr[1] += ptr_inc[1];
-                array_idx[1] = static_cast<size_t>(array_idx[1] + arrayStep[1]);
+                array_idx[1] = static_cast<size_t>(CPLUnsanitizedAdd<GUInt64>(
```

Comment:
Couldn't use do the add with size_t type and avoid the extra cast?

Reply:
Also, if an overflow is OK here, a comment would be good.

Reply:
Cast logic simplified, and following comment added
// CPLUnsanitizedAdd needed as arrayStep[] might be negative, and
// thus automatic conversion from negative to big unsigned might
// occur